src/lualdap.c

```diff
@@ -981,8 +981,12 @@ static int lualdap_open_simple (lua_State *L) {
 	if (strstr(host, "://") != NULL) {
 		err = ldap_initialize(&conn->ld, host);
 	} else {
-		const char *host_with_schema = lua_pushfstring(L, "ldap://%s", host);
-		err = ldap_initialize(&conn->ld, host_with_schema);
+		lua_getfield (L, 1, "gsub");
```

Comment:
Why do you call `gsub` on the argument, instead of `luaL_gsub`? I would think the latter is guaranteed to give you the results you expect, while the former could be modified by the user?

Reply:
`luaL_gsub` doesn't handle the *regex* stuff like *class character* and *capture*.
see https://github.com/lua/lua/blob/master/lauxlib.c#L975-L995
`luaL_gsub` does only string substitution.

Now, I retrieve the method `:gsub`, but I could use the function `string.gsub` instead.

Reply:
I would prefer that, if I was to write this code. That makes it easier for you to know what you have available and weird changes a user makes cannot throw a wrench into your gears. But what is the consensus of the Lua community on this?